mantle/platform/cluster.go

```diff
@@ -221,25 +221,22 @@ enabled = false`, 0644)
 	}
 
 	if bc.bf.baseopts.OSContainer != "" {
-		if bc.Distribution() != "rhcos" {
-			return nil, fmt.Errorf("oscontainer is only supported on the rhcos distribution")
-		}
-		conf.AddSystemdUnitDropin("pivot.service", "00-before-sshd.conf", `[Unit]
-Before=sshd.service`)
-		conf.AddSystemdUnit("pivot.service", "", platformConf.Enable)
-		conf.AddSystemdUnit("pivot-write-reboot-needed.service", `[Unit]
-Description=Touch /run/pivot/reboot-needed
-ConditionFirstBoot=true
+		conf.AddSystemdUnit("kola-container-rebase.service", fmt.Sprintf(`[Unit]
+Description=Rebase to target container
+ConditionPathExists=!/etc/kola-rebase-done
+Before=sshd.service
+Wants=network-online.target
+After=network-online.target
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/mkdir -p /run/pivot
```

Comment:
This might be off base, but would it make sense to move the unit into a separate file? It feels like we would want to encourage our unit to be changed without effecting the cluster.go code fundamentally right?

Open to being completely wrong lol.

Reply:
Yes, though I think what we really want to do is support this officially in butane - we're doing something here that we expect users of [layering](https://github.com/coreos/enhancements/blob/main/os/coreos-layering.md) to do as well!


So OK to keep as is for now and logically move the script there later?

Reply:
Ah ok, yeah absolutely, def not a blocker.